packages/compiler/lib/decorators.tsp

```diff
@@ -409,7 +409,7 @@ enum BytesKnownEncoding {
 extern dec encode(
   target: Scalar | ModelProperty,
   encoding: string | EnumMember,
-  encodedAs?: string | numeric
+  encodedAs?: Reflection.Scalar
```

Comment:
```suggestion
  encodedAs?:Scalar
```

think we actually can just have `Scalar` here like above

Reply:
I guess the question is, is this they way we should go? I don't feel like the error message is very logical.

Reply:
I mean maybe but I think that's out of the scope of this PR. We can file a new issue to have a better error message when checking something is assignable to a reflection type.

Reply:
Okay, I've gone this way and updated that test to throw "invalid-argument" instead of "invalid-encode".